lib/CryptoNoteCore/Currency.cpp

```diff
@@ -268,8 +268,11 @@ bool Currency::constructMinerTx(
     const AccountPublicAddress &minerAddress,
     Transaction &tx,
     const BinaryArray &extraNonce /* = BinaryArray()*/,
-    size_t maxOuts /* = 1*/) const
+    size_t maxOuts /* = 1*/,
+    uint64_t blockTarget /* = 0*/) const
 {
+    if (blockTarget == 0)
+        blockTarget = difficultyTarget();
```

Comment:
In case that the selfish miner post two or more blocks consecutively with the same time then this will be a problem. I would recommend removing line 274 and 275. If we need these lines, please let us know the reason. Thanks.

Reply:
I see,  I've missed the fact,  that blockTarget can be zero in real call. It is attempt to implement default value for blockTarget equal to `difficultyTarget()`. More direct approach is set default value to `difficultyTarget()` instead of `0`in method definition, but defaultTarget is the method of the same class,  so I've decided to implement it this way to not use constructions like  `uint64_t blockTarget = this->defaultTarget()` in the header file in method definition. 

I can change default value to `0xffffffff` for example (and change this lines to 
```if (blockTarget == 0xfffffffff) blockTarget = difficultyTarget();```),  is it ok?

Reply:
Okay. Makes Sense.